packages/pwa-kit-dev/src/ssr/server/loading-screen/index.html

```diff
@@ -86,8 +86,9 @@ <h1 class="loading-text">Building your app</h1>
                 }
 
                 waitForReady().then(() => {
-                    // Redirect to homepage
-                    window.location = url.origin
+                    // Redirect to homepage if path is not defined
+                    const path = url.searchParams.get('path') ?? ''
+                    window.location = url.origin + path
```

Comment:
```suggestion
                    window.location = new URL(path, url.origin)
```
[minor] This does a bit of extra validation of the path.
(It could technically allow a specially crafted URL to redirect to a different origin, but I don't think that's a real concern.)